core/dbt/version.py

```diff
@@ -27,81 +51,175 @@ def get_latest_version(version_url: str = PYPI_VERSION_URL):
     return dbt.semver.VersionSpecifier.from_version_string(version_string)
 
 
-def get_installed_version():
-    return dbt.semver.VersionSpecifier.from_version_string(__version__)
+def _get_core_msg(installed, latest) -> Tuple[str, str]:
+    core_msg_lines, update_info = _get_core_msg_lines(installed, latest)
+    return (
+        _format_core_msg(core_msg_lines),
+        update_info,
+    )
 
 
-def get_package_pypi_url(package_name: str) -> str:
-    return f"https://pypi.org/pypi/dbt-{package_name}/json"
+def _get_core_msg_lines(installed, latest) -> Tuple[List[List[str]], str]:
+    installed_s = installed.to_version_string(skip_matcher=True)
+    installed_line = ["installed", installed_s, ""]
+    update_info = ""
+
+    if latest is None:
+        update_info = (
+            "The latest version of dbt could not be determined!\n"
+            "Make sure that the following URL is accessible:\n"
+            f"{PYPI_VERSION_URL}"
+        )
+        return [installed_line], update_info
 
+    latest_s = latest.to_version_string(skip_matcher=True)
+    latest_line = ["latest", latest_s, green("Up to date!")]
 
-def get_version_information():
-    flags.USE_COLORS = True if not flags.USE_COLORS else None
+    if installed > latest:
+        latest_line[2] = green("Ahead of latest version!")
+    elif installed < latest:
+        latest_line[2] = yellow("Update available!")
+        update_info = (
+            "Your version of dbt is out of date! "
```

Comment:
[Not blocking, cosmetic only]
```suggestion
            "Your version of dbt-core is out of date! "
```